### PR TITLE
Add BLS Signature Verification for Multiple Payloads

### PR DESCRIPTION
**Pull request by tholonious**

### CODE REVIEW

**Comment by tholonious** [comment]
Current dependencies on/for this PR:
* feature/proof-of-stake
  * **PR #570** <a href="https://app.graphite.dev/github/pr/deso-protocol/core/570" target="_blank">Graphite</a> 
    * **PR #575** <a href="https://app.graphite.dev/github/pr/deso-protocol/core/575" target="_blank">Graphite</a>  👈

This comment was auto-generated by [Graphite](https://app.graphite.dev/github/pr/deso-protocol/core/575?utm_source=stack-comment).

File: bls/signature.go [hunk 1]
```diff
@@ -11,20 +11,21 @@ import (
...
 	flowCrypto "github.com/onflow/flow-go/crypto"
+	"github.com/onflow/flow-go/crypto/hash"
 )
 
-// The SigningAlgorithm for BLS keys is BLSBLS12381 which is BLS on the BLS 12-381 curve.
+// The signingAlgorithm for BLS keys is BLSBLS12381 which is BLS on the BLS 12-381 curve.
 // This is the only supported BLS signing algorithm in the flowCrypto package.
 // BLS is used such that we can aggregate signatures into one signature.
-const SigningAlgorithm = flowCrypto.BLSBLS12381
+const signingAlgorithm = flowCrypto.BLSBLS12381
```
Comment (tholonious) [inline bls/signature.go:20]: This `signingAlgorithm` const and `hashingAlgorithm` global variable are implementation details of how this package wraps the flow BLS scheme implementation underneath. 

We don't want this package to expose them externally, so it's best to make them private.

File: bls/signature.go [hunk 2]
```diff
@@ -41,15 +42,30 @@ func AggregateSignatures(signatures []*Signature) (*Signature, error) {
...
 	}
-	return flowCrypto.VerifyBLSSignatureOneMessage(flowPublicKeys, signature.flowSignature, payloadBytes, HashingAlgorithm)
+	return flowCrypto.VerifyBLSSignatureOneMessage(flowPublicKeys, signature.flowSignature, payloadBytes, hashingAlgorithm)
+}
+
+// VerifyAggregateSignatureMultiplePayloads takes in a slice of bls.PublicKeys, a bls.Signature, and a slice of payloads.
+// It returns true if each bls.PublicKey at index i has signed its respective payload at index i in the payloads slice.
+// The input bls.Signature is the aggregate signature of each public key's partial bls.Signatures for its respective payload.
+func VerifyAggregateSignatureMultiplePayloads(publicKeys []*PublicKey, signature *Signature, payloadsBytes [][]byte) (bool, error) {
+	var flowPublicKeys []flowCrypto.PublicKey
```
Comment (lazynina) [inline bls/signature.go:60]: do we want a simple validation that `len(publicKeys) == len(payloadsBytes)`?

File: bls/signature.go [hunk 3]
```diff
@@ -41,15 +42,30 @@ func AggregateSignatures(signatures []*Signature) (*Signature, error) {
...
+	for _, publicKey := range publicKeys {
+		flowPublicKeys = append(flowPublicKeys, publicKey.flowPublicKey)
+	}
```
Comment (lazynina) [inline bls/signature.go:63]: Starting to see this repeat. I think it's okay to have it duplicated for now, but if we continue to convert slices of PublicKey to slices of flowCrypto.PublicKey, we should write a little helper function.

File: bls/signature.go (hunk 2 above)
Comment (tholonious) [reply bls/signature.go:60]: Good call. Definitely worth adding it here

File: bls/signature.go (hunk 3 above)
Comment (tholonious) [reply bls/signature.go:63]: Good diea. Seems appropriate to create the helper function now.

**Review by lazynina — APPROVED** [review]

**Pull request merged**